test/unittests/eof_validation_test.cpp

```diff
@@ -1166,3 +1172,12 @@ TEST_F(eof_validation, EOF1_unreferenced_subcontainer_valid)
     const auto embedded = eof_bytecode(bytecode{OP_INVALID});
     add_test_case(eof_bytecode(OP_STOP).container(embedded), EOFValidationError::success);
 }
+
+TEST_F(eof_validation, max_nested_containers)
+{
+    bytecode code = eof_bytecode(OP_INVALID);
+    while (code.size() <= std::numeric_limits<uint16_t>::max())
```

Comment:
It still can contain more nested containers, only limited by the maximum size of the main container.

For example: `while (code.size() <= std::numeric_limits<uint16_t>::max() + 9)` can still pass this test.

Reply:
Not sure what you mean, with `+9` it's the same nesting level (2621) and container size (65545)

Subcontainer size have to be `<= std::numeric_limits<uint16_t>::max()` because it's size is encoded as 2 bytes in parent header.